cmd/thanos/query.go

```diff
@@ -690,7 +696,11 @@ func runQuery(
 	}
 
 	engineFactory := apiv1.NewQueryEngineFactory(
-		engineOpts,
+		engine.Opts{
+			EngineOpts:       engineOpts,
+			Engine:           promql.NewEngine(engineOpts),
+			EnableSubqueries: enableThanosEngineSubqueries,
```

Comment:
Feel like it is an overkill to me to have a flag for it

Reply:
I added it by popular request to have this feature configurable ( for a short time ). Once this has proven stable enough we can probably remove the flag again.

Reply:
Having said that im fairly confident that it works like in upstream promql since it passes acceptance tests and fuzzing

Reply:
:heavy_plus_sign: maybe we can just enable subqueries? The more people exposed to this, the faster bugs will be fixed (if any)

Reply:
Can we have it for a short time and we run a nightly build and once we are happy ill take out the switch again?

Reply:
I guess we could use prometheus engine in case of issues too, so maybe its not so bad.

Reply:
Consulted with a teammate; ill take out the flag in a couple hours!